### ISSUE TITLE
Uninformative integration testing failures

### ISSUE THREAD

**Issue by natanlao** [BD2KGenomics/toil#1700]
Exhibit A: http://jenkins.cgcloud.info/job/toil-pull-request-integration-tests/1145/console (#1572 4f52f1f02c8afdbf1c304e1b377c220fa76fd201)
Exhibit B: http://jenkins.cgcloud.info/job/toil-pull-request-integration-tests/1039/console (#1506 84807b483eb4e7b81c4c60f51d4d84be3a1bf948)

Jenkins recognizes that testing suite fails but cannot identify which test failed. No traceback in console output.

In the case of Exhibit B, suite fails because of infinite loop that causes suite to time out - the problematic test, however, is not identified.

**Comment by jessebrennan** [comment]
More examples:
- http://jenkins.cgcloud.info/job/toil-pull-request-integration-tests/1172/console (#1392)
- http://jenkins.cgcloud.info/job/toil-pull-request-integration-tests/1176/console (#1572)
- http://jenkins.cgcloud.info/job/toil-pull-request-integration-tests/1177/console (#1724)
- http://jenkins.cgcloud.info/job/toil-pull-requests/2815/console (#1668)

Is this happening on all branches?

This is show-stopping for me as is. I'm not really sure what causes this to happen or how to go about trying to fix it.

**Comment by jessebrennan** [comment]
No longer show stopping. I think I have found the source of the problem in at least some of the cases. If when running integration tests on specifies only a particular test such as 
```
$ make integration_test tests=src/toil/test/utils/utilsTest.py::UtilsTest::testAWSProvisionerUtils
```
the test may pass successfully, but `make` will fail overall. I suspect that this is because multiple threads are launched to run each test from a list within run_tests.py and since only one test is specified, the threads with nothing to do fail.

This theory also makes sense with another one of the cases I was experiencing the error. I had commented out one of the tests and `make` would "fail" despite none of the tests failing.
